lang/java/avro/src/main/java/org/apache/avro/file/DataFileReader.java

```diff
@@ -170,7 +171,7 @@ public void seek(long position) throws IOException {
     vin = DecoderFactory.get().binaryDecoder(this.sin, vin);
     datumIn = null;
     blockRemaining = 0;
-    blockStart = position;
+    blockFinished();
```

Comment:
This has the same result, but I'm using `blockFinished()` to use the same code in all cases.